chia/wallet/wallet_node.py

```diff
@@ -385,8 +385,9 @@ async def _start_with_fingerprint(
         self.wallet_state_manager.state_changed("sync_changed")
 
         # Populate the balance caches for all wallets
-        for wallet_id in self.wallet_state_manager.wallets:
-            await self.get_balance(wallet_id)
+        async with self.wallet_state_manager.lock:
+            for wallet_id in self.wallet_state_manager.wallets:
+                await self._update_balance_cache(wallet_id)
```

Comment:
this seems to change more than what you describe in the PR. You no longer call `self._balance_cache.get(wallet_id, Balance())` from here now. You also don't take `self.wallet_state_manager.sync_mode` into account.

Reply:
> this seems to change more than what you describe in the PR.

Okay yeah i slightly changed the "Purpose" section.

 > You no longer call `self._balance_cache.get(wallet_id, Balance())` from here now. 

Its not relevant since this is only to populate the caches on startup. 

> You also don't take `self.wallet_state_manager.sync_mode` into account.

This is also not relevant for startup since its only to make the RPC calls use the cache while we are syncing and update the balance if not.